addons/repair/models/sale_order.py

```diff
@@ -70,7 +70,7 @@ def write(self, vals_list):
         old_product_uom_qty = {line.id: line.product_uom_qty for line in self}
         res = super().write(vals_list)
         for line in self:
-            if line.state in ('sale', 'done'):
+            if line.state in ('sale', 'done') and line.product_uom:
                 if float_compare(old_product_uom_qty[line.id], 0, precision_rounding=line.product_uom.rounding) <= 0 and float_compare(line.product_uom_qty, 0, precision_rounding=line.product_uom.rounding) > 0:
                     self._create_repair_order()
                 if float_compare(old_product_uom_qty[line.id], 0, precision_rounding=line.product_uom.rounding) > 0 and float_compare(line.product_uom_qty, 0, precision_rounding=line.product_uom.rounding) <= 0:
```

Comment:
This logic could be skipped if `product_uom_qty` is not in `vals` (to be renamed from vals_list)

Reply:
The 'done' state on SO's doesn't exist anymore (and even when it did, it shouldn't have triggered such business flows)